mava/wrappers/gym.py

```diff
@@ -0,0 +1,403 @@
+        agents_view, reward, terminated, truncated, info = self._env.step(actions)
+
+        self.running_count_episode_return += float(np.mean(reward))
+        self.running_count_episode_length += 1
+
+        metrics = {
+            "episode_return": self.running_count_episode_return,
+            "episode_length": self.running_count_episode_length,
+            "is_terminal_step": np.logical_or(terminated, truncated).all().item(),
+        }
+        if "won_episode" in info:
+            metrics["won_episode"] = info["won_episode"]
+
+        info["metrics"] = metrics
+
+        return agents_view, reward, terminated, truncated, info
+
+
+class GymAgentIDWrapper(gymnasium.Wrapper):
+    """Add one hot agent IDs to observation."""
+
+    def __init__(self, env: gymnasium.Env):
+        super().__init__(env)
+
+        self.agent_ids = np.eye(self.env.num_agents)
+        self.observation_space = self.modify_space(self.env.observation_space)
+
+    def reset(
+        self, seed: Optional[int] = None, options: Optional[dict] = None
+    ) -> Tuple[NDArray, Dict]:
+        """Reset the environment."""
+        obs, info = self.env.reset(seed, options)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, info
+
+    def step(self, action: list) -> Tuple[NDArray, float, bool, bool, Dict]:
+        """Step the environment."""
+        obs, reward, terminated, truncated, info = self.env.step(action)
+        obs = np.concatenate([self.agent_ids, obs], axis=1)
+        return obs, reward, terminated, truncated, info
+
+    def modify_space(self, space: spaces.Space) -> spaces.Space:
+        if isinstance(space, spaces.Box):
+            new_shape = (space.shape[0], space.shape[1] + len(self.agent_ids))
+            return spaces.Box(
+                low=space.low[0][0], high=space.high[0][0], shape=new_shape, dtype=space.dtype
+            )
+        elif isinstance(space, spaces.Tuple):
+            return spaces.Tuple(self.modify_space(s) for s in space)
+        else:
+            raise ValueError(f"Space {type(space)} is not currently supported.")
+
+
+class GymToJumanji:
+    """Converts from the Gym API to the Jumanji API."""
+
+    def __init__(self, env: gymnasium.vector.VectorEnv):
+        self.env = env
+        self.single_action_space = env.unwrapped.single_action_space
+        self.single_observation_space = env.unwrapped.single_observation_space
+
+    def reset(self, seed: Optional[list[int]] = None, options: Optional[dict] = None) -> TimeStep:
+        obs, info = self.env.reset(seed=seed, options=options)  # type: ignore
+
+        num_agents = len(self.env.single_action_space)  # type: ignore
+        num_envs = self.env.num_envs
+
+        ep_done = np.zeros(num_envs, dtype=float)
+        rewards = np.zeros((num_envs, num_agents), dtype=float)
+        teminated = np.zeros(num_envs, dtype=float)
+
+        timestep = self._create_timestep(obs, ep_done, teminated, rewards, info)
```

Comment:
Small bug here, this should return `step_type.First`, but it won't. I'd say just create the timestep manually here